### DISCUSSION TITLE
Start and restart individual tasks in turbo run

### DISCUSSION THREAD

**Discussion by cmaycumber** [vercel/turborepo#7487] [Ideas]
### Which project is this feature idea for?

Turborepo

### Describe the feature you'd like to request

Currently I've been using the library: `https://www.npmjs.com/package/run-pty` to start all tasks in parallel but also allows restarting individual tasks, and providing input to individual tasks via the console. 

This would be helpful because it allows developers to run a large amount of tasks in parallel but if one fails or stalls and can be restarted without having to restart all the packages in the monorepo.  

### Describe the solution you'd like

I'd like `turbo run` to have the ability to restart individual tasks and accept standard input for individual tasks. 

### Describe alternatives you've considered

Using `run-pty` instead of `turbo run` for development.

**Comment by mehulkar** [comment]
Closing in favor of https://github.com/vercel/turbo/issues/986 and https://github.com/vercel/turbo/issues/1235

**Comment by valeriangalliat** [comment]
Those 2 issues don't seem to be related? Watch mode and forward stdin are nice to have but it doesn't seem that both of those would allow starting and restarting individual tasks in turbo run

**Comment by mehulkar** [comment]
hey @valeriangalliat I'll re-open this, because this issue is larger in scope compared to Watch Mode and forwarding stdin, but wanted to explain my thinking. The original issue description 

> allows restarting individual tasks

Watch Mode is about re-running individual tasks when a related file or env var changes. I felt like that's one of the primary reasons to "restart" a task, but you're right that it's not exactly the same as a persistent server that needs to be restarted.

> providing input to individual tasks via the console

This is about forwarding stdin, but `run-pty` is more extensive (with a dashboard UI to select which task gets stdin), etc. I imagine that that kind of interface will be a second iteration of forwarding stidn!

**Comment by anthonyshew** [comment]
Watch Mode is now available in Turborepo 2.0: https://turbo.build/repo/docs/reference/watch

**Comment by IAmNatch** [comment]
WatchMode is great, but doesn't account for the situation I'm experiencing:
1. The task has a built in watcher, but it crashes. - In this case, I don't want watch mode to restart on every file change, but I do want to be able to manually restart after the task crashes.
2. The task has a built in watcher, but I'm worried that I changed something that is outside of the watchers scope (like a config file), and therefore I want to restart the task.

Sure, I can use `ctl-c` and re-run all tasks, but this is more expensive that restarting a single task. So it would be a nice addition to the "More binds" menu "Restart task".
